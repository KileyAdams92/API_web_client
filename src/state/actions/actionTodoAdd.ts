import { ActionType } from "../../types/ActionType"
import { makeActionCreator } from "../../utils/makeActionCreator"
import { EntityTodo } from "../../types/EntityTodo"

export const actionTodoAdd = makeActionCreator<EntityTodo>(ActionType.TodoAdd)
export type ActionTodoAdd = ReturnType<typeof actionTodoAdd>
