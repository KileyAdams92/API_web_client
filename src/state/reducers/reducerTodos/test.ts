import { actionTodoAdd } from "../../actions/actionTodoAdd"
import { reducerTodos } from "."

describe("reducerTodos", () => {
  it("handles actionTodoAdd", () => {
    expect(
      reducerTodos(
        undefined,
        actionTodoAdd({ id: "1", title: "Test Todo", complete: true }),
      ),
    ).toMatchSnapshot()
  })
})
