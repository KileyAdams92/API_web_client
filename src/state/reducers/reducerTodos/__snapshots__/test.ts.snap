// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducerTodos handles actionTodoAdd 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "complete": true,
      "id": "1",
      "title": "Test Todo",
    },
  },
  "result": Array [
    "1",
  ],
}
`;
